go/mysql/endtoend/replication_test.go

```diff
@@ -908,6 +908,10 @@ func TestRowReplicationTypes(t *testing.T) {
 		t.Fatal(err)
 	}
 	defer dConn.Close()
+	// We have tests for zero dates, so we need to allow that for this session.
+	if _, err := dConn.ExecuteFetch("SET @@session.sql_mode=REPLACE(REPLACE(@@session.sql_mode, 'NO_ZERO_DATE', ''), 'NO_ZERO_IN_DATE', '')", 0, false); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This is now needed since the tests try to insert zero dates and we need to test those, so they need to be allowed for this session here.